actionpack/lib/action_dispatch/system_testing/server.rb

```diff
@@ -0,0 +1,23 @@
+require "rack/handler/puma"
+
+module ActionDispatch
+  module SystemTesting
+    class Server # :nodoc:
+      def run
+        register
+        setup
+      end
+
+      private
+        def register
+          Capybara.register_server :rails_puma do |app, port, host|
+            Rack::Handler::Puma.run(app, Port: port, Threads: "0:1")
```

Comment:
Can we add an error message if Puma is not installed, so users can add it to their Gemfile